### DISCUSSION TITLE
Adding a Timeframe for Persistent High Alert

### DISCUSSION THREAD

**Discussion by Seithx** [NightscoutFoundation/xDrip#3753] [Ideas]
Couldnt find a thread about this.

I have highs at specific times at night and I get up at 12 AM to inject.

I turned this feature on, but, during the day I am awake to monitor and it does  bother me during meetings.

![Screenshot_20241103-183759](https://github.com/user-attachments/assets/834d9647-291b-44bb-a01d-698f4a18c106)
](url)

**Comment by Navid200** [comment]
Would this work out if you could choose a volume profile for the persistent high alert?

**Reply by Seithx** [reply]
as a workaround, perhaps, but if I have to change it every time, its not practical - during night i want it blaring out, during day, not at all.

**Reply by Navid200** [reply]
That's what the ascending volume profile is for:
https://navid200.github.io/xDrip/docs/Ascending-volume-profile.html  

When you are in a meeting, it starts at very low volume.  You can even set it to start only vibrating with 0 volume for sound.  You can then snooze it before it makes any sound.

**Reply by Seithx** [reply]
let me try this. I will report back.

**Reply by Navid200** [reply]
The persistent high alert does not respect the volume profile currently.
It is something I would like to add.  I am sorry I was not clear about that.

**Reply by Teute0815** [reply]
I support that request. It's the same thing I just messaged you a few hours ago @Navid200 and which I requested here #2091 

> My individual example:
I'd love to have a persitent high alert, if I'm above 10 mmol for 60 minutes from 01 a.m. to 08 a.m.

That would need individual thresholds (not pulled from "Glucose Levels") and a timeplan.
Instead of it being an individual setting, It would be great to include it in the "Glucose Level Alerts" (if that's possible), as a potential checkbox. That way it could incorporate all the different individual options of those alerts.

**Reply by Navid200** [reply]
The persistent high alert already has a threshold that is independent of the High Value.  
This was added only a few Nightlies ago.  So, it is very recent.

It is not going to have a time of day added by me.  

It will have a new change where it will respect the volume profile so that you can choose ascending volume profile.  However, that is not easy and I am not sure when I will be able to accomplish it.

**Reply by Seithx** [reply]
Well, it did help as a workaround to me not being able to snooze properly, etc, but on Shabbath it drove me nuts - it rang for 10 hours, not even sure why since it started happening after i left, then got back, and stopped for an hour or two during the day. Anyways, I believe the timeframe would be a game changer, regardless of Shabbath, as a way to nudge those highs that may happen at night and i dont wake up.

**Reply by Navid200** [reply]
@Seithx I don't follow.
If you don't wake up with the alert at night, how would adding a time frame wake you up?!

**Reply by Seithx** [reply]
> @Seithx I don't follow. If you don't wake up with the alert at night, how would adding a time frame wake you up?!

by it ringging non stop until I do... the regular alerts stop, because i like them during the day to ring once every 35 mins.

the time frame as I said is very important, different times, different needs.

**Reply by Navid200** [reply]
I wonder if you can put yourself in my shoes.
If xDrip as is can accomplish what you need, there is no need to change xDrip.
Even though what you suggest may give you what you need, that's not enough to justify adding it to xDrip.

If you want your Glucose Level alert to ring until you stop it, all xDrip glucose level alerts do that already.
I don't see how adding time of day is going to make any difference in that respect.

**Comment by Seithx** [comment]
Xdrip does do what I want, but it only does it binary - either black or white - ring constantly, or don't ring at all, basically....

I can suggest another option, as the current settings do not allow for it to stop unless I am able to stop it.

adding an 'auto disable' where I don't want to be notified about any alert, for example, Saturday? I keep forgetting, but cannot shut it down.

**Reply by Navid200** [reply]
I prefer not to add day of week or time of day to xDrip.  
I know that it has been added to the glucose level alert.  I consider that unfortunate.
The danger is one week you may decide to do things differently than your usual on a Saturday but forget to adjust xDrip accordingly.  

There has to be a balance in how much customization we offer.
Too much of it makes xDrip extremely complicated.
I am trying to clean it up.  But, it is an uphill battle.  
Look at this where I am trying to remove a redundant setting: https://github.com/NightscoutFoundation/xDrip/pull/3071  
It has been sitting there for a year and a half.  

You wouldn't believe how many times people ask why a setting is buried several levels.  
We should only add new settings when absolutely necessary.  

xDrip is designed for a diabetic who carries the phone all the time.  It is not meant for the phone to be left at home expecting xDrip to sit quietly.

**Comment by bobemmanoonan-blip** [comment]
I'd appreciate this to have fewer alarms during day, but ability to manage son's bg overnight and sleep. But get this is more customisation. Xdrip kicks way more arse than commercial apps, so thank you.

**Reply by Navid200** [reply]
Deciding what should change and what should stay the same is harder than making a change.

The more options there are in an app, the less useful those options become.  Eventually, no one will use any of those options and will look for a simpler app.  If we make xDrip an app that is full of settings, we could effectively make it useless for users.  

If I have been high for a long time, I need to know.  I don't understand why that would be different during the day versus during the night.
Would you please help me understand why that could make a difference?

**Reply by bobemmanoonan-blip** [reply]
I agree with your sentiment that adding more options will make the app more scary and thereby less usable. The ability to customise alarms is the biggest benefit of xdrip. The persistent high alarm is unique and really useful. The difference between day and night is that I'm regularly looking at my sons during the day and dont need more alarms interrupting work etc, whereas at night I'm trying to sleep but not wanting him to be high all night. Pre discovering persistent alarm i spent a few nights awake.The fewer alarms the better. This is a nice to have for me. I'll live with the persistent alarms during the day.
